tests/Feature/Api/ApplicationControllerTest.php

```diff
@@ -110,4 +111,28 @@ public function testUpdateBasicsValidation(): void
         $response->assertStatus(422);
         $response->assertJsonFragment(['message' => 'The given data was invalid.']);
     }
+
+    public function testUpdateApplicationStep(): void
+    {
+        $applicant = factory(Applicant::class)->create();
+        $application = factory(JobApplication::class)->states('draft')->create([
+            'applicant_id' => $applicant->id
+        ]);
+        $step = Step::where('name', 'basic')->first();
+        $application->attachSteps();
+        $application->refresh();
+
+
+        $data = [
+            'applicationId' => $application->id,
+            'stepId' => $step->id,
+        ];
+
+        $response = $this->actingAs($applicant->user)
+        ->json('put', "$this->baseUrl/applications/$application->id/steps/$step->id", $data);
+        $response->assertOk();
+
+        $application->refresh();
+        $response->assertJsonFragment($application->applicationTimelineSteps());
```

Comment:
Suggestions for more test cases:
- Assert that the 'basic' step is touched, while the others are not. The current assertion would pass if the request returned the steps without changing any.
- Test that calling the request on the same step multiple times doesn't do anything weird.